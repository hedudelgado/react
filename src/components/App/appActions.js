import { HeaderInterface, InvalidSessionErrorCodeMap, InvalidSessionHelper } from 'mega-code';
import {
  getCountries,
  getNewsletterRegions,
  getNewsletterPreferences,
  getProfile,
  updateProfile,
  deleteCustomer,
} from '../../utils/api';

// actions
export const USER_LOGGED_IN_SUCCESSFULLY = 'USER_LOGGED_IN_SUCCESSFULLY';
export const COUNTRIES_RECEIVED = 'COUNTRIES_RECEIVED';
export const PROFILE_RECEIVED = 'PROFILE_RECEIVED';
export const REGIONS_RECEIVED = 'REGIONS_RECEIVED';
export const NEWSLETTER_RECEIVED = 'NEWSLETTER_RECEIVED';
export const SET_EDIT_VIEW = 'SET_EDIT_VIEW';
export const SET_NOTIFICATION_MESSAGE = 'SET_NOTIFICATION_MESSAGE';
export const PROFILE_UPDATED = 'PROFILE_UPDATED';
export const NEWSLETTER_UPDATED = 'NEWSLETTER_UPDATED';
export const RESET_NEWSLETTER_INFO = 'RESET_NEWSLETTER_INFO';
export const RESET_USER_INFO = 'RESET_USER_INFO';
// views
export const DEFAULT_VIEW = undefined;
export const USER_DETAILS_VIEW = 'USER_DETAILS_VIEW';
export const PASSWORD_VIEW = 'PASSWORD_VIEW';
export const PAYMENT_CARD_VIEW = 'PAYMENT_CARD_VIEW';
export const ADD_GUEST_VIEW = 'ADD_GUEST_VIEW';
export const REGULAR_GUESTS_VIEW = 'REGULAR_GUESTS_VIEW';
export const NEWSLETTER_VIEW = 'NEWSLETTER_VIEW';
export const ROOM_REQUIREMENTS_VIEW = 'ROOM_REQUIREMENTS_VIEW';
export const EXTRAS_PREFERENCES_VIEW = 'EXTRAS_PREFERENCES_VIEW';
export const MEMORABLE_WORD_VIEW = 'MEMORABLE_WORD_VIEW';
// errors
export const PROFILE_ERROR = 'PROFILE_ERROR';
export const PASSWORD_ERROR = 'PASSWORD_ERROR';
export const NEWSLETTER_ERROR = 'NEWSLETTER_ERROR';
export const PAYMENT_CARD_ERROR = 'PAYMENT_CARD_ERROR';
export const USER_DETAILS_ERROR = 'USER_DETAILS_ERROR';
export const REGULAR_GUESTS_ERROR = 'REGULAR_GUESTS_ERROR';
export const MEMORABLE_WORD_ERROR = 'MEMORABLE_WORD_ERROR';
export const ROOM_REQUIREMENTS_ERROR = 'ROOM_REQUIREMENTS_ERROR';
export const EXTRAS_PREFERENCES_ERROR = 'EXTRAS_PREFERENCES_ERROR';

const checkForInvalidSession = (error, config, dispatch) => {
  if (InvalidSessionHelper
    .isSessionExpired(error, config)) {
    dispatch({
      type: RESET_USER_INFO,
    });
    dispatch({
      type: RESET_NEWSLETTER_INFO,
    });
    HeaderInterface.userLogOut(HeaderInterface.LOG_OUT_REASONS.SESSION_EXPIRED);
  }
};

export const getSettings = (email, sessionId, business) => (dispatch, getState) => {
  getProfile(email, sessionId, business).then((profile) => {
    dispatch({
      type: PROFILE_RECEIVED,
      profile,
    });
  }).catch((error) => {
    checkForInvalidSession(error, InvalidSessionErrorCodeMap.ACCOUNT.invalidSession, dispatch);
    dispatch({
      type: PROFILE_ERROR,
      error,
    });
  });
  getCountries()
    .then((countries) => {
      dispatch({
        type: COUNTRIES_RECEIVED,
        countries,
      });
    });
  if (getState().application.showEmailPreferences) {
    getNewsletterRegions()
      .then((regions) => {
        dispatch({
          type: REGIONS_RECEIVED,
          regions,
        });
      });

    getNewsletterPreferences(email)
      .then((newsletter) => {
        dispatch({
          type: NEWSLETTER_RECEIVED,
          newsletter,
        });
      })
      .catch(e => dispatch({
        type: NEWSLETTER_ERROR,
        error: e,
      }));
  }
};

export const updateUserProfile = payload => dispatch => updateProfile(payload).catch((error) => {
  checkForInvalidSession(error, InvalidSessionErrorCodeMap.ACCOUNT.invalidSession, dispatch);
  return Promise.reject(error);
});

export const deleteUserProfile = payload => dispatch => deleteCustomer(payload).catch((error) => {
  checkForInvalidSession(error, InvalidSessionErrorCodeMap.DELETE_PROFILE.invalidSession, dispatch);
  return Promise.reject(error);
});

export default {
  getSettings,
};
