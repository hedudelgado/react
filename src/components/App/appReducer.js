import {
  SET_NOTIFICATION_MESSAGE, SET_EDIT_VIEW, DEFAULT_VIEW,
} from './appActions';

const defaultState = {
  editViewName: DEFAULT_VIEW,
};

export default (state = defaultState, action) => {
  switch (action.type) {
    case SET_EDIT_VIEW:
      return {
        ...state,
        editViewName: action.editViewName,
      };
    case SET_NOTIFICATION_MESSAGE:
      return {
        ...state,
        notificationMessage: action.notificationMessage,
      };
    default:
      return state;
  }
};
