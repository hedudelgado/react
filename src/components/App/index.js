export { default } from './App';

export { default as AppActions } from './appActions';

export { default as AppContainer } from './AppContainer';

export { default as modalReducer } from './modalReducer';
