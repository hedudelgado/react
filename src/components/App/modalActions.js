export const MODAL_OPEN = 'MODAL_OPEN';
export const MODAL_CLOSE = 'MODAL_CLOSE';

export const openModal = content => ({
  type: MODAL_OPEN,
  content,
});

export const closeModal = () => ({
  type: MODAL_CLOSE,
});
