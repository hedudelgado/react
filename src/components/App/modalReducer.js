import {
  MODAL_OPEN,
  MODAL_CLOSE,
} from './modalActions';

export default (state = {
  display: false,
  content: null,
}, action) => {
  const {
    type,
    content,
  } = action;

  switch (type) {
    case MODAL_OPEN:
      return {
        ...state,
        display: true,
        content,
      };
    case MODAL_CLOSE:
      return {
        ...state,
        display: false,
        content: null,
      };
    default:
      return state;
  }
};
