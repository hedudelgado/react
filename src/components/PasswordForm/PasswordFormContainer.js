import React, { PureComponent, createRef } from 'react';
import { func, object } from 'prop-types';
import { Formik } from 'formik';
import BaseStyle from '../../layout/BaseStyle';
import Typography from '../../layout/Typography';
import PasswordForm from './PasswordForm';
import Notification from '../../layout/Notification';
import { validateCurrentPassword, validateNewPassword, validateConfirmPassword } from '../../utils/validations';
import {
  DEFAULT_VIEW,
  PASSWORD_ERROR,
  PASSWORD_VIEW,
} from '../App/appActions';
import { apiErrorCodes } from '../../config.json';

const validatePasswordForm = (values) => {
  const errors = {};

  errors.currentPassword = validateCurrentPassword(values.currentPassword);
  errors.newPassword = validateNewPassword(values);
  errors.confirmPassword = validateConfirmPassword(values);

  Object.keys(errors).forEach((error) => {
    if (errors[error] === undefined) delete errors[error];
  });

  return errors;
};

export default class PasswordFormContainer extends PureComponent {
  constructor(props) {
    super(props);

    this.form = createRef();
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(values, actions) {
    const {
      profile,
      setNotificationMessage,
      setEditView,
      updateProfile,
    } = this.props;
    const {
      newPassword,
      currentPassword: password,
    } = values;
    const profileRequest = { ...profile, password, newPassword };

    updateProfile(profileRequest).then(() => {
      setNotificationMessage(PASSWORD_VIEW);
      setEditView(DEFAULT_VIEW);
      actions.setSubmitting(false);
    }).catch(({ code }) => {
      const { INCORRECT_PASSWORD } = apiErrorCodes;
      if (code === INCORRECT_PASSWORD) {
        actions.setFieldError('currentPassword', 'password.error.incorrect');
      } else {
        setNotificationMessage(PASSWORD_ERROR);
      }
    });
  }

  render() {
    const {
      setEditView,
      dictionary,
      app: { notificationMessage },
    } = this.props;

    return (
      <BaseStyle component="section" className="component-container">
        {notificationMessage === PASSWORD_ERROR && (
          <Notification data-test="error-notification" priority={2} variant="error" ariaLive="polite" show>
            {dictionary['generic.error']}
          </Notification>
        )}
        <Typography component="h4" className="pi-color-greyscale-darkest">
          {dictionary['password.changepassword.title']}
        </Typography>

        <Formik
          data-test="password-form-wrapper"
          ref={this.form}
          initialValues={{ currentPassword: '', newPassword: '', confirmPassword: '' }}
          validate={validatePasswordForm}
          onSubmit={this.handleSubmit}
          render={formikProps => (
            <PasswordForm
              {...formikProps}
              data-test="password-form"
              dictionary={dictionary}
              setEditView={setEditView}
            />
          )}
        />
      </BaseStyle>
    );
  }
}

PasswordFormContainer.propTypes = {
  dictionary: object.isRequired,
  setEditView: func.isRequired,
  profile: object.isRequired,
  app: object.isRequired,
  setNotificationMessage: func.isRequired,
  updateProfile: func.isRequired,
};
