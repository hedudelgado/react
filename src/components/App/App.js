import React, { Component } from 'react';
import {
  array,
  object,
  func,
  bool,
  node,
} from 'prop-types';
import {
  PASSWORD_VIEW,
  PAYMENT_CARD_VIEW,
  USER_DETAILS_VIEW,
  ROOM_REQUIREMENTS_VIEW,
  EXTRAS_PREFERENCES_VIEW,
} from './appActions';
import { ComponentTogglerContainer } from '../ComponentToggler';
import UserProfile from '../UserProfile';
import Password from '../Password';
import UserProfileFormContainer from '../UserProfileForm';
import PasswordFormContainer from '../PasswordForm';
import { PaymentCardContainer } from '../PaymentCard';
import { RegularGuestsContainer } from '../RegularGuests';
import { NewsletterContainer } from '../Newsletter';
import Modal from '../../layout/Modal';
import PaymentCardFormContainer from '../PaymentCardForm';
import { DeleteCustomerContainer } from '../DeleteCustomer';
import RoomRequirements from '../RoomRequirements';
import RoomRequirementsFormContainer from '../RoomRequirementsForm';
import ExtrasPreferences from '../ExtrasPreferences';
import ExtrasPreferencesFormContainer from '../ExtrasPreferencesForm';
import BaseStyle from '../../layout/BaseStyle';
import MemorableWord from '../MemorableWord';
import MoveAccount from '../MoveAccount';
import {
  STAYER, SELF, managePermissionFlag, USER_LOGGED_IN_EVENT,
} from '../../config.json';
import ManagePermission from '../ManagePermission';

class App extends Component {
  constructor(props) {
    super(props);
    this.handleLoggedIn = this.handleLoggedIn.bind(this);
  }

  componentDidMount() {
    window.addEventListener('message', this.handleLoggedIn);
  }

  componentWillUnmount() {
    window.removeEventListener('message', this.handleLoggedIn);
  }

  async handleLoggedIn(event) {
    if (event.data.action === USER_LOGGED_IN_EVENT) {
      const { getSettings } = this.props;
      const { contactDetail: { email }, sessionId, business } = JSON.parse(event.data.profile);
      getSettings(email, sessionId, !!business);
    }
  }

  render() {
    const {
      countries,
      dictionary,
      profile,
      regions,
      newsletter,
      displayModal,
      handleModalClose,
      modalContent,
      userDataReceived,
      application: {
        showEmailPreferences,
      },
    } = this.props;

    const {
      business,
      business: {
        accessLevel = '',
        tetheredGuid = '',
      } = {},
    } = profile;

    const appReady = userDataReceived
      && (!showEmailPreferences || !!regions.length)
      && !!countries.length
      && (!showEmailPreferences || !!Object.keys(newsletter).length)
      && !!Object.keys(dictionary).length
      && !!Object.keys(profile).length;

    const showExtrasPreferences = !(accessLevel === SELF || accessLevel === STAYER);

    return (
      <div data-test="AccountSettings">
        {appReady
        && (
          <BaseStyle className="wb-typography app-container">
            <ComponentTogglerContainer
              viewName={USER_DETAILS_VIEW}
              view={UserProfile}
              edit={UserProfileFormContainer}
            />
            <ComponentTogglerContainer
              viewName={PASSWORD_VIEW}
              view={Password}
              edit={PasswordFormContainer}
            />
            <ComponentTogglerContainer
              viewName={PAYMENT_CARD_VIEW}
              view={PaymentCardContainer}
              edit={PaymentCardFormContainer}
            />
            {managePermissionFlag && (
              <ManagePermission dictionary={dictionary} />
            )}
            <RegularGuestsContainer />
            {showEmailPreferences && (
              <NewsletterContainer />
            )}
            <ComponentTogglerContainer
              viewName={ROOM_REQUIREMENTS_VIEW}
              view={RoomRequirements}
              edit={RoomRequirementsFormContainer}
            />
            {showExtrasPreferences && (
              <ComponentTogglerContainer
                viewName={EXTRAS_PREFERENCES_VIEW}
                view={ExtrasPreferences}
                edit={ExtrasPreferencesFormContainer}
              />
            )}
            {!!tetheredGuid && (
              <MemorableWord />
            )}
            {business && !tetheredGuid && (
              <MoveAccount />
            )}
            {!business && (
            <DeleteCustomerContainer />
            )}
            {displayModal && <Modal data-test="Modal" display content={modalContent} handleClose={handleModalClose} />}
          </BaseStyle>
        )
      }
      </div>
    );
  }
}

App.propTypes = {
  countries: array.isRequired,
  dictionary: object.isRequired,
  profile: object.isRequired,
  getSettings: func.isRequired,
  displayModal: bool.isRequired,
  handleModalClose: func.isRequired,
  modalContent: node,
  newsletter: object.isRequired,
  regions: array.isRequired,
  userDataReceived: bool.isRequired,
  application: object.isRequired,
};

export default App;
