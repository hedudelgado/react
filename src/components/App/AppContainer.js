import { connect } from 'react-redux';
import { getSettings as getSettingsAction } from './appActions';
import { closeModal } from './modalActions';

import App from './App';

const mapStateToProps = ({
  modal: {
    display: displayModal,
    content: modalContent,
  },
  dictionary,
  config,
  countries,
  regions,
  newsletter,
  application,
  userProfile: {
    userDataReceived,
    profile,
  },
}) => ({
  profile,
  displayModal,
  modalContent,
  dictionary,
  config,
  countries,
  regions,
  newsletter,
  application,
  userDataReceived,
});

const mapDispatchToProps = dispatch => ({
  getSettings:
    (email, sessionId, business) => dispatch(getSettingsAction(email, sessionId, business)),
  handleModalClose: () => dispatch(closeModal()),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
