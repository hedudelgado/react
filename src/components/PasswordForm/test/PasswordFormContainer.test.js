import React from 'react';
import { shallow } from 'enzyme';
import PasswordFormContainer from '../PasswordFormContainer';
import { validateCurrentPassword, validateNewPassword, validateConfirmPassword } from '../../../utils/validations';
import { PASSWORD_ERROR } from '../../App/appActions';

jest.mock('../../../utils/validations', () => ({
  __esModule: true,
  validateCurrentPassword: jest.fn(),
  validateNewPassword: jest.fn(),
  validateConfirmPassword: jest.fn(),
}));

describe('PasswordFormContainer Component', () => {
  let wrapper;
  let props;
  const updateProfile = jest.fn(() => Promise.resolve({}));

  beforeEach(() => {
    props = {
      dictionary: {
        'password.button.update': 'Update',
        'password.button.cancel': 'Cancel',
      },
      setEditView: jest.fn(),
      profile: {},
      setNotificationMessage: jest.fn(),
      app: {},
      updateProfile,
    };
    wrapper = shallow(<PasswordFormContainer {...props} />);
  });

  describe('Renders PasswordFormContainer Element', () => {
    it('should render the component', () => {
      expect(wrapper).not.toBeEmptyRender();
    });

    it('should match snapshot', () => {
      expect(wrapper.find('Formik').dive()).toMatchSnapshot();
    });

    it('should display an error message when state.showError is true', () => {
      const newProps = {
        ...props,
        app: { notificationMessage: PASSWORD_ERROR },
      };
      wrapper = shallow(<PasswordFormContainer {...newProps} />);
      expect(wrapper.find('[data-test="error-notification"]')).toHaveLength(1);
    });

    it('should not display an error message when state.showError is false', () => {
      wrapper.setState({ showError: false });
      expect(wrapper.find('[data-test="error"]')).toHaveLength(0);
    });

    it('should execute togglePasswordView when clicking the cancel update button', () => {
      wrapper
        .find('[data-test="password-form-wrapper"]')
        .dive()
        .find('[data-test="password-form"]')
        .props()
        .setEditView();
      expect(props.setEditView).toHaveBeenCalledTimes(1);
    });
  });

  describe('Password Form submission', () => {
    const mockSubmitOptions = (setFieldError = jest.fn(), setSubmitting = jest.fn()) => {
      const submitValues = {
        currentPassword: 'Password1',
        newPassword: 'Password134',
        confirmPassword: 'Password134',
      };
      const submitActions = { setFieldError, setSubmitting };

      wrapper
        .find('[data-test="password-form-wrapper"]')
        .simulate('submit', submitValues, submitActions);
    };

    it('should execute password update success', async () => {
      mockSubmitOptions();

      await Promise.all([updateProfile]).then(() => {
        expect(props.setNotificationMessage).toHaveBeenCalledTimes(1);
      });
    });

    it('should show incorrect password message', async () => {
      const setFieldError = jest.fn();

      updateProfile.mockImplementationOnce(() => Promise.reject({
        code: '012',
        details: ['Existing password doesn\'t match records'],
      }));

      mockSubmitOptions(setFieldError);

      await Promise.all([updateProfile]).then(() => {
        expect(setFieldError).toHaveBeenCalledTimes(1);
      });
    });

    it('should set ERROR notification on generic API failure', async () => {
      updateProfile.mockImplementationOnce(() => Promise.reject({
        code: '999',
        details: ['Bad Request [400]'],
      }));

      mockSubmitOptions();

      await Promise.all([updateProfile]).then(() => {
        expect(props.setNotificationMessage)
          .toHaveBeenCalledTimes(1)
          .toHaveBeenCalledWith(PASSWORD_ERROR);
      });
    });
  });

  describe('Password Form validations', () => {
    const values = {
      currentPassword: 'Premier1',
      newPassword: 'Premier2',
      confirmPassword: 'Premier2',
    };

    it('should validate all fields calling all validation functions with the correct values', () => {
      wrapper
        .find('[data-test="password-form-wrapper"]')
        .props()
        .validate(values);

      expect(validateCurrentPassword).toHaveBeenNthCalledWith(1, 'Premier1');
      expect(validateNewPassword).toHaveBeenNthCalledWith(1, {
        currentPassword: 'Premier1',
        newPassword: 'Premier2',
        confirmPassword: 'Premier2',
      });
      expect(validateConfirmPassword).toHaveBeenNthCalledWith(1, {
        currentPassword: 'Premier1',
        newPassword: 'Premier2',
        confirmPassword: 'Premier2',
      });
    });

    it('should return the correct validation result based on correct values', () => {
      const validationResult = wrapper
        .find('[data-test="password-form-wrapper"]')
        .props()
        .validate(values);

      expect(validationResult).toEqual({
        currentPassword: undefined,
        newPassword: undefined,
        confirmPassword: undefined,
      });
    });

    it('should return the correct validation result based on incorrect values', () => {
      validateCurrentPassword.mockImplementationOnce(() => 'You need to enter your password');
      validateNewPassword.mockImplementationOnce(() => 'Incorrect new password');
      validateConfirmPassword.mockImplementationOnce(() => 'Passwords do not match');

      const validationResult = wrapper
        .find('[data-test="password-form-wrapper"]')
        .props()
        .validate(values);

      expect(validationResult).toEqual({
        currentPassword: 'You need to enter your password',
        newPassword: 'Incorrect new password',
        confirmPassword: 'Passwords do not match',
      });
    });
  });
});
