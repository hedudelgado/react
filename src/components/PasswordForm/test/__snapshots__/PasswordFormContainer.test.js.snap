// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordFormContainer Component Renders PasswordFormContainer Element should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "dirty": false,
      "errors": Object {},
      "handleBlur": [Function],
      "handleChange": [Function],
      "handleReset": [Function],
      "handleSubmit": [Function],
      "initialValues": Object {
        "confirmPassword": "",
        "currentPassword": "",
        "newPassword": "",
      },
      "isSubmitting": false,
      "isValid": false,
      "isValidating": false,
      "registerField": [Function],
      "resetForm": [Function],
      "setError": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFormikState": [Function],
      "setStatus": [Function],
      "setSubmitting": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "status": undefined,
      "submitCount": 0,
      "submitForm": [Function],
      "touched": Object {},
      "unregisterField": [Function],
      "validate": [Function],
      "validateField": [Function],
      "validateForm": [Function],
      "validateOnBlur": true,
      "validateOnChange": true,
      "validationSchema": undefined,
      "values": Object {
        "confirmPassword": "",
        "currentPassword": "",
        "newPassword": "",
      },
    }
  }
>
  <PasswordForm
    data-test="password-form"
    dictionary={
      Object {
        "password.button.cancel": "Cancel",
        "password.button.update": "Update",
      }
    }
    dirty={false}
    errors={Object {}}
    handleBlur={[Function]}
    handleChange={[Function]}
    handleReset={[Function]}
    handleSubmit={[Function]}
    initialValues={
      Object {
        "confirmPassword": "",
        "currentPassword": "",
        "newPassword": "",
      }
    }
    isSubmitting={false}
    isValid={false}
    isValidating={false}
    registerField={[Function]}
    resetForm={[Function]}
    setEditView={[MockFunction]}
    setError={[Function]}
    setErrors={[Function]}
    setFieldError={[Function]}
    setFieldTouched={[Function]}
    setFieldValue={[Function]}
    setFormikState={[Function]}
    setStatus={[Function]}
    setSubmitting={[Function]}
    setTouched={[Function]}
    setValues={[Function]}
    submitCount={0}
    submitForm={[Function]}
    touched={Object {}}
    unregisterField={[Function]}
    validateField={[Function]}
    validateForm={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    values={
      Object {
        "confirmPassword": "",
        "currentPassword": "",
        "newPassword": "",
      }
    }
  />
</ContextProvider>
`;
